app/concepts/chatbot/brain.rb

```diff
@@ -0,0 +1,137 @@
+      @msg  = msg
+      @fbid = msg.sender.id
+    end
+
+    def hear
+      case @msg
+      when MessengerClient::Message::Optin    then handle_optin
+      when MessengerClient::Message::Text     then handle_text_message
+      when MessengerClient::Message::Postback then handle_postback
+      else
+        Rails.logger.warn "Chatbot::Brain#hear doesn't know how to handle: #{@msg.inspect}"
+        random_message("chatbot.responses.dont_understand")
+        send_messages("chatbot.responses.help", help_url: help_url)
+      end
+    end
+
+    ##########################
+    # Handlers
+    ##########################
+
+    def handle_text_message
+      return handle_unregistered_user if user.nil?
+      case @msg.text
+      when list_matcher("chatbot.user.help", locale)
+        send_messages("chatbot.responses.help", help_url: help_url)
+      when list_matcher("chatbot.user.hello", locale)
+        random_message("chatbot.responses.hello")
+      else
+        random_message("chatbot.responses.dont_understand")
+      end
+    end
+
+    def handle_postback
+      return handle_unregistered_user if user.nil?
+      case @msg.postback.to_s
+      when Postbacks::GET_STARTED then
+        opts = {
+          w2h_url:           Rails.application.routes.url_helpers.root_url,
+          registration_url:  Rails.application.routes.url_helpers.new_user_registration_url,
+          notifications_url: Rails.application.routes.url_helpers.users_notifications_url,
+        }
+        send_messages("chatbot.responses.get_started", opts)
+      when Postbacks::HELP_PAYLOAD then send_messages("chatbot.responses.help", help_url: help_url)
+      else random_message("chatbot.responses.dont_understand")
+      end
+    end
+
+
+    def handle_optin
+      ChatbotOperation::UserSignUp.(@msg)
+      first_name  = user.first_name
+      message     = I18n.t("chatbot.responses.onboarding", locale: user.locale, first_name: first_name, help_url: help_url)
+      MultiMessageJob.perform_later(fbid, message)
+    end
+
+    def handle_unregistered_user
+      info_url          = Rails.application.routes.url_helpers.root_url
+      registration_url  = Rails.application.routes.url_helpers.new_user_registration_url
+      notifications_url = Rails.application.routes.url_helpers.users_notifications_url
+      message = I18n.t("chatbot.responses.please_register",
+                       locale:            profile.w2h_locale,
+                       first_name:        profile.first_name,
+                       notifications_url: notifications_url,
+                       info_url:          info_url,
+                       registration_url:  registration_url)
+      MultiMessageJob.perform_later(fbid, message)
+    end
+
+    ##########################
+    # Message Utils
+    ##########################
+
+    def list_matcher(locale_key, l, only_at_start = true)
+      /#{only_at_start ? "^" : ""}(#{I18n.t(locale_key, locale: l).join("|")})/i
```

Comment:
11. Should we [escape](https://ruby-doc.org/core-2.2.0/Regexp.html#method-c-escape) the locale strings when constructing the regexp? (e.g. what if it contains a "?" character, that would otherwise be interpreted as the regex-optional-operator, no?)